pkg/controller/kuberneteseventsource/reconcile_test.go

```diff
@@ -0,0 +1,379 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Veroute.on 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberneteseventsource
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp"
+	sourcesv1alpha1 "github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	controllertesting "github.com/knative/eventing-sources/pkg/controller/testing"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/record"
+	"testing"
+)
+
+var (
+	trueVal   = true
+	targetURI = "http://sinkable.sink.svc.cluster.local/"
+)
+
+const (
+	image               = "github.com/knative/test/image"
+	sourceName          = "testkuberneteseventsource"
+	testNS              = "testnamespace"
+	testSAN             = "testserviceaccount"
+	sourceUID           = "2a2208d1-ce67-11e8-b3a3-42010a8a00af"
+	deployGeneratedName = "" //sad trombone
+
+	sinkableDNS = "sinkable.sink.svc.cluster.local"
+
+	sinkableName       = "testsink"
+	sinkableKind       = "Sink"
+	sinkableAPIVersion = "duck.knative.dev/v1alpha1"
+
+	unsinkableName       = "testunsinkable"
+	unsinkableKind       = "KResource"
+	unsinkableAPIVersion = "duck.knative.dev/v1alpha1"
+
+	sinkServiceName       = "testsinkservice"
+	sinkServiceKind       = "Service"
+	sinkServiceAPIVersion = "v1"
+)
+
+// Adds the list of known types to Scheme.
+func duckAddKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(
+		duckv1alpha1.SchemeGroupVersion,
+		&duckv1alpha1.AddressableTypeList{},
+	)
+	metav1.AddToGroupVersion(scheme, duckv1alpha1.SchemeGroupVersion)
+	return nil
+}
+
+func init() {
+	// Add types to scheme
+	sourcesv1alpha1.SchemeBuilder.AddToScheme(scheme.Scheme)
+	duckv1alpha1.AddToScheme(scheme.Scheme)
+	duckAddKnownTypes(scheme.Scheme)
+}
+
+var testCases = []controllertesting.TestCase{
+	{
+		Name:       "existing ready container source",
+		Reconciles: &sourcesv1alpha1.KubernetesEventSource{},
+		InitialState: []runtime.Object{
+			func() runtime.Object {
+				s := getSource()
+				s.UID = sourceUID
+				return s
+			}(),
+			func() runtime.Object {
+				u := getContainerSource_isReady()
+				u.SetOwnerReferences(getOwnerReferences())
+				return u
+			}(),
+		},
+		ReconcileKey: fmt.Sprintf("%s/%s", testNS, sourceName),
```

Comment:
I recommend removing `ReconcileKey` and `IgnoreTimes` out of every test case and just set them in the test itself, because every test case uses the same values for those two fields.